File: ivalia_mobile/lib/api/models/response/api_response.dart
class ApiResponse {
  int type;
  String? messages;
  Object? data;

  ApiResponse({this.type = 0, this.messages, this.data});

  factory ApiResponse.fromJson(Map<String, dynamic> json) {
    return ApiResponse(
      type: json['type'] as int,
      messages: json['messages'] as String?,
      data: json['data'] as Object?,
    );
  }

  bool isSuccessful() => type == ResponseType.success;
}

class ResponseType {
  static const int success = 0;
  static const int error = 1;
}

File: Ivalia_WebAPI/DataTransfer/DTO/Response/ApiResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Ivalia.DataTransfer.DTO.Response
{
    public class ApiResponse
    {
        public int Type { get; set; }
        public string Messages { get; set; }
        public object Data { get; set; }

        public bool IsSuccessful() => Type == Status.SUCCESS;

        public static class Status
        {
            public static int SUCCESS = 0;
            public static int ERROR = 1;
        }
    }
}
